File: treebolicOwlServices/src/main/java/org/treebolic/owl/service/Deployer.java
/*
 * Copyright (c) Treebolic 2019. Bernard Bou <1313ou@gmail.com>
 */

package org.treebolic.owl.service;

import java.io.File;
import java.io.InputStream;

import androidx.annotation.NonNull;

/**
 * Data deployer
 *
 * @author Bernard Bou
 */
@SuppressWarnings("WeakerAccess")
public class Deployer
{
	/**
	 * Directory path
	 */
	static private final String PATH = "treebolic";

	/**
	 * Parent directory
	 */
	@NonNull
	private final File dir;

	/**
	 * Constructor
	 *
	 * @param parentDir parent directory
	 */
	public Deployer(final File parentDir)
	{
		this.dir = new File(parentDir, Deployer.PATH);
	}

	/**
	 * Process download stream (such as expanding download zipped stream, copy file, etc)
	 *
	 * @param fin file input stream
	 * @return directory
	 */
	@NonNull
	public File process(final InputStream fin)
	{
		return this.dir;
	}
}

File: treebolicOwlServices/src/main/java/org/treebolic/owl/Deployer.kt
/*
 * Copyright (c) Treebolic 2023. Bernard Bou <1313ou@gmail.com>
 */
package org.treebolic.owl

import java.io.File
import java.io.InputStream

/**
 * Data deployer
 *
 * @param parentDir parent directory
 *
 * @author Bernard Bou
 */
class Deployer(parentDir: File?) {

    /**
     * Parent directory
     */
    private val dir: File = File(parentDir, PATH)

    /**
     * Process download stream (such as expanding download zipped stream, copy file, etc)
     *
     * @param fin file input stream
     * @return directory
     */
    fun process(fin: InputStream?): File {
        return dir
    }

    companion object {

        /**
         * Directory path
         */
        private const val PATH = "treebolic"
    }
}
